### PR TITLE
HDS-1431 Add possibility to use external label for Select and Combobox components

### PR DESCRIPTION
**Pull request by VilleMiekkoja**
## Description

Add possibility to use external label for Select and Combobox components. This enables custom labels that are placed and styled differently than the components own label.

## Related Issue

https://helsinkisolutionoffice.atlassian.net/browse/HDS-1431

## How Has This Been Tested?

- Locally in dev machine
- Automated visual regression tests

👉 [Combobox demo](https://city-of-helsinki.github.io/hds-demo/external-label-for-dropdowns/?path=/story/components-dropdowns-combobox--with-external-label)
👉 [Select demo](https://city-of-helsinki.github.io/hds-demo/external-label-for-dropdowns/?path=/story/components-dropdowns-select--with-external-label)

## Screenshots (if appropriate):

image

### CODE REVIEW

File: packages/react/src/components/dropdown/select/Select.tsx [hunk 1]
```diff
@@ -65,6 +65,7 @@ export interface SelectCustomTheme {
 }
 
 export type CommonSelectProps<OptionType> = {
+  'aria-labelledby'?: string;
```
Comment (minevala) [inline packages/react/src/components/dropdown/select/Select.tsx:68]: Should we have some description text for this property?

File: packages/react/src/components/dropdown/select/Select.tsx [hunk 2]
```diff
@@ -116,7 +117,7 @@ export type CommonSelectProps<OptionType> = {
   /**
    * The label for the dropdown
    */
-  label: React.ReactNode;
+  label?: React.ReactNode;
```
Comment (minevala) [inline packages/react/src/components/dropdown/select/Select.tsx:120]: Could we have some guidance or typing that component requires either label or aria-labelledby?

**Review by minevala — COMMENTED** [review]
A nice addition 👍 
I was thinking should we rather have a functional test for component than regression tests?
The feature feels more like a functional addition rather than a visual change even tough it hides the default label. And the functional tests are way faster

File: packages/react/src/components/dropdown/select/Select.tsx (hunk 1 above)
Comment (VilleMiekkoja) [reply packages/react/src/components/dropdown/select/Select.tsx:68]: - [ ] Add description for aria-labelledby property

File: packages/react/src/components/dropdown/select/Select.tsx (hunk 2 above)
Comment (VilleMiekkoja) [reply packages/react/src/components/dropdown/select/Select.tsx:123]: - [ ] Add typing for conditionally requiring eitgher label or aria-labelledby

**Comment by VilleMiekkoja** [comment]
> A nice addition 👍 I was thinking should we rather have a functional test for component than regression tests? The feature feels more like a functional addition rather than a visual change even tough it hides the default label. And the functional tests are way faster

- [ ] Add unit test for the case without label (with aria-labelledby)

**Comment by VilleMiekkoja** [comment]
Unicus tested this and it should be ok.

**Review by minevala — APPROVED** [review]
🏅

**Pull request merged**